python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -1525,14 +1525,52 @@ def mexists(
         n_chunks = 0
         for chunk in chunk_iterable(refs, chunk_size=chunk_size):
             chunk_result = self._mexists(chunk, artifact_existence)
-            if log.isEnabledFor(VERBOSE):
-                n_results = len(chunk_result)
-                n_checked += n_results
+
+            # The log message level and content depend on how many
+            # datasets we are processing.
+            n_results = len(chunk_result)
+
+            log_threshold = VERBOSE
+            n_checked += n_results
+
+            # This sum can take some time so only do it if we know the
+            # result is going to be used.
+            n_found = 0
+            if log.isEnabledFor(log_threshold):
                 # Can treat the booleans as 0, 1 integers and sum them.
                 n_found = sum(chunk_result.values())
                 n_found_total += n_found
-                log.verbose(
-                    "Number of datasets found in datastore for chunk %d = %d/%d (running total: %d/%d)",
+
+            # We are deliberately not trying to count the number of refs
+            # provided in case it's in the millions. This means there is a
+            # situation where the number of refs exactly matches the chunk
+            # size and we will switch to the multi-chunk path even though
+            # we only have a single chunk.
+            if n_results < chunk_size and n_chunks == 0:
+                # Single chunk will be processed so we can provide more detail.
+                if n_results == 1:
+                    ref = list(chunk_result)[0]
+                    log.log(
+                        log_threshold,
```

Comment:
It's possible this should be a DEBUG message since we don't log anything for the simple `datastore.exists()` API. The  multi-chunk case if VERBOSE because it takes a significant amount of time to run and it's good to have feedback. The middle ground is the single chunk and whether the verbose vs debug depends on the amount of time that it took to do the existence check (9999 datasets being checked will obviously take much longer than 2 datasets).

Reply:
I don't have strong opinion, but I'm a big fan of trying to keep things simple. 🙂 Logging everything at VERBOSE may be OK, as I think most people will run it with INFO.